code/go/0chain.net/chaincore/chain/handler.go

```diff
@@ -1585,6 +1622,62 @@ func (c *Chain) MinerStatsHandler(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func txnIterHandlerFunc(c *Chain, w http.ResponseWriter) func(context.Context, datastore.CollectionEntity) bool {
+	return func(ctx context.Context, ce datastore.CollectionEntity) bool {
+		txn, ok := ce.(*transaction.Transaction)
+		if !ok {
+			logging.Logger.Error("generate block (invalid entity)", zap.Any("entity", ce))
+			return false
+		}
+
+		s, err := c.GetStateById(c.GetLatestFinalizedBlock().ClientState, txn.ClientID)
```

Comment:
We should get the latest finalized block once and use it instead of getting it on each iteration calling. The block will be changed overtime.

Reply:
Sure. I'll make the changes